client/src/modules/Registry/pages/Student/index.tsx

```diff
@@ -26,24 +27,40 @@ export type Props = {
   session: Session;
 };
 
+interface IdName {
+  id: number;
+  name: string;
+}
+
 export function StudentRegistry(props: Props & { courseAlias?: string }) {
+  const { githubId } = props.session;
   const [form] = Form.useForm();
   const update = useUpdate();
   const [loading, setLoading] = useState(false);
   const [submitted, setSubmitted] = useState(false);
   const [courses, setCourses] = useState([] as Course[]);
+  const [registeredForCourses, setRegisteredForCourses] = useState([] as IdName[]);
   const [location, setLocation] = useState(null as Location | null);
   const [initialData, setInitialData] = useState({} as Partial<UserFull>);
 
   useAsync(async () => {
```

Comment:
Also maybe please consider moving it to a separate hook. It looks too complicated now for a component

Reply:
done